### DISCUSSION TITLE
How would you set minimum & maximum of items?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#857] [Q&A] [answered]
Hi

I thought about min and max items in a list:
Wouldn't it be nice to have such a feature?
Something like `mv-list-min="1"` `mv-list-max="5"`

I tried to do it with expressions, but my brain get inside a horrible loop :(
Do we have a demo somewhere with this use case?
or how would you set it?

**Comment by DmitrySharabin** [comment, marked as answer]
Will something I use in the [Interactive carousel widget](https://codepen.io/dmitrysharabin/pen/jjKwGG) help? There, I limited the maximum number of images in the carousel by hiding the Add button.

**Reply by GalinhaLX** [reply]
Do you mean using:
```
mv-editor-min="3"
mv-editor-max="10"
```

Are those mavo functions?

I also tried something [here](https://codepen.io/galinhalx/pen/NWyaYGd), based on a previous codepen you provided

**Reply by GalinhaLX** [reply]
oh, I see, using

```
<aside hidden="[imageCount >= maxCount]">
				<button class="mv-add-image">
					+ Add image
				</button>
			</aside>
```

Joins the idea on my [codepen](https://codepen.io/galinhalx/pen/NWyaYGd), but I was not able to make it work. 
Am I missing something?

**Reply by DmitrySharabin** [reply]
Will this be helpful?
https://codepen.io/dmitrysharabin/pen/KKQXRxj

> Joins the idea on my [codepen](https://codepen.io/galinhalx/pen/NWyaYGd), but I was not able to make it work.
Am I missing something?

I noticed a couple of issues:
- the delete button in the item bar has the `mv-delete` class not `mv-remove`
- when using expressions in attributes, don't forget `[` and `]` around the expression. E.g., that won't work without brackets `style="if(count(item.$all) > 2,'display:none','')"`
- there is no need to provide the third argument of the `if()` function if it's empty. E.g., you may simplify the above expression: `style="[if(count(item.$all) > 2,'display:none')]"`.

Hope it helps!

**Reply by GalinhaLX** [reply]
Oh, you're absolutly right !

I'm using `mv-delete`, `mv-add`, and `mv-move` for item's bar
and `mv-add-item` for bottom add button

I forgot [ and ] around the expression 😱, that's why all my attemps where not working

Thank you @DmitrySharabin , it really helps: I also need more coffee maybe 😁

**Reply by DmitrySharabin** [reply]
😂 You are always welcome!

**Reply by GalinhaLX** [reply]
Oh
I see you've been doing a [codepen](https://codepen.io/dmitrysharabin/pen/KKQXRxj) also :)

Better than mine! so I'll take yours if you don't mind

Except I changed a little thing `>= `instead of just `>`, otherwise you'll have 2 delete buttons on first item (after adding a few items)
`mv-if="count(item.$all) >= min_count"`
<img width="609" alt="image" src="https://user-images.githubusercontent.com/89095737/170096489-a1e5dc81-c191-41c6-99b0-ab223f613801.png">


Also saw that user can add 1 extra item, due to the way mavo is done:
- if you reach maximum items, you still can delete the content of item text (by pressing delete keyboard) 
- if you just delete the content, without deleting the item itself: you'll see the '+ add' button appearing

![image](https://user-images.githubusercontent.com/89095737/170096108-df479987-cce8-4de8-8e45-927661d28ed8.png)


is there a way for mavo to make a test of an empty value?

**Reply by DmitrySharabin** [reply]
> so I'll take yours if you don't mind

Of course, I don't mind! :)

> is there a way for mavo to make a test of an empty value

If I'm not mistaken, all properties without values get the `mv-empty` class you can use as a hook in your CSS.

**Reply by GalinhaLX** [reply]
Super cool ! Thanks a lot @DmitrySharabin 

Sharing the updated [codepen](https://codepen.io/galinhalx/pen/VwQMErE)
